docs/configuration/configuration.md

```diff
@@ -100,6 +100,7 @@ An example [oauth2-proxy.cfg]({{ site.gitweb }}/contrib/oauth2-proxy.cfg.example
 | `-request-logging-format` | string | Template for request log lines | see [Logging Configuration](#logging-configuration) |
 | `-resource` | string | The resource that is protected (Azure AD only) | |
 | `-reverse-proxy` | bool | are we running behind a reverse proxy, controls whether headers like X-Real-Ip are accepted | false |
+| `-real-client-ip-header` | string | Header used to determine the real IP of the client, has no effect without the `-reverse-proxy` (one of: X-Forwarded-For, X-Real-IP, or X-ProxyUser-IP) | X-Real-IP |
```

Comment:
```suggestion
| `-real-client-ip-header` | string | Header used to determine the real IP of the client, requires `-reverse-proxy` to be set (one of: X-Forwarded-For, X-Real-IP, or X-ProxyUser-IP) | X-Real-IP |
```

Reply:
I do feel like this implies an error will be produced if this option is specified without `-reverse-proxy`—which is not true—but maybe it should be?

Reply:
That's a fantastic idea! Could you look into updating the option validation to error if this is set but reverse proxy isn't? You would need to look at the flag to see if the value was changed, maybe that's not so easy

If it looks nasty then maybe we don't proceed with that, but I'm happy with how the phrasing I've suggested either way, unless one of the other maintainers disagrees cc @steakunderscore

Reply:
Hmm, I'm not sure there's a not-totally-hacky way to determine whether or not -real-client-ip-header has been specified without significant code-change.  Additionally, I'm not sure it does make sense to crash if it's specified without `-reverse-proxy`, as it breaks the source-agnostic nature of the configuration: the behaviour changes whether it's set to `X-Real-IP` on the command line or from the default value.

I'll just update the language, and if @steakunderscore thinks we should change the behaviour here I can look into it more.